service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -54,25 +54,25 @@ public AssessmentQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTempla
                 "A.ftmaxitems AS fieldTestMaxItems, \n" +
                 "S.name AS subject, \n" +
                 "A.virtualtest AS assessmentKey, \n" +
-                "P.propname, \n" +
-                "P.propvalue, \n" +
-                "P.propdescription, \n" +
                 "CT.ftstartdate, \n" +
                 "CT.ftenddate, \n" +
                 "CT.accommodationfamily, \n" +
                 "CT.maxopportunities, \n" +
                 "CT.abilityslope, \n" +
-                "CT.abilityintercept \n" +
+                "CT.abilityintercept, \n" +
+                "SP.ftstartdate AS segFieldTestStartDate, \n" +
+                "SP.ftenddate AS segFieldTestEndDate \n" +
             "FROM itembank.tblsetofadminsubjects A \n " +
-            "JOIN configs.client_testproperties CT ON CT.testid = A.testid \n " +
-            "  OR CT.testid = (\n " +
-            "    select parentTsa.testid from itembank.tblsetofadminsubjects tsa \n " +
-            "    join itembank.tblsetofadminsubjects parentTsa on tsa.virtualtest = parentTsa._key\n " +
-            "    where tsa._key = A._key \n " +
-            "  ) \n " +
+            "JOIN configs.client_testproperties CT \n" +
+            "   ON (CT.testid = A.testid AND CT.clientname = :clientName) \n " +
+            "   OR (CT.testid = (\n " +
+            "    SELECT parentTsa.testid FROM itembank.tblsetofadminsubjects tsa \n " +
+            "    JOIN itembank.tblsetofadminsubjects parentTsa ON tsa.virtualtest = parentTsa._key\n " +
+            "    WHERE tsa._key = A._key \n " +
+            "  ) AND CT.clientname = :clientName) \n " +
```

Comment:
I'm fine with saying the best practices we should try to follow is to only include the joining keys in the `JOIN` statement and any filters in the `WHERE` clause.